flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java

```diff
@@ -140,7 +136,11 @@ private void openCli(Executor executor) {
 
         try (CliClient cli = new CliClient(terminalFactory, executor, historyFilePath)) {
             if (options.getInitFile() != null) {
-                boolean success = cli.executeInitialization(readFromURL(options.getInitFile()));
+                if (isApplicationMode(executor.getSessionConfig())) {
+                    throw new SqlClientException(
+                            "Sql Client doesn't support to run init files when deploying script into cluster.");
+                }
+                boolean success = cli.executeInitialization(options.getInitFile());
```

Comment:
I'm confuse that what content are init file? is it  sql query, sql ddl or config option? After see the line 147 and line 153 error message, I guess it is a sql query.

Reply:
You can take a look at th doc. 

> Script file that used to init the session context. If ge error in execution, the sql client will exit. Notice it's not allowed to add query or insert into the init file.